test/mempool-test.js

```diff
@@ -233,6 +233,63 @@ describe('Mempool', function() {
     }));
   });
 
+  it('should get spent coins and reflect in coinview', async () => {
+    const wallet = new MemWallet();
+    const addr = wallet.getAddress();
+
+    const dummyCoin = dummyInput(addr, random.randomBytes(32));
+
+    const mtx1 = new MTX();
+    mtx1.addOutput(wallet.getAddress(), 50000);
+    mtx1.addCoin(dummyCoin);
+
+    wallet.sign(mtx1);
+
+    const tx1 = mtx1.toTX();
+    const coin1 = Coin.fromTX(tx1, 0, -1);
+
+    const mtx2 = new MTX();
+    mtx2.addOutput(wallet.getAddress(), 10000);
+    mtx2.addOutput(wallet.getAddress(), 30000); // 10k fee
+    mtx2.addCoin(coin1);
+
+    wallet.sign(mtx2);
+
+    const tx2 = mtx2.toTX();
+
+    await mempool.addTX(tx1);
+
+    {
+      const view = await mempool.getCoinView(tx2);
+      const sview = await mempool.getCoinView(tx2);
+      assert(view.hasEntry(coin1));
+      assert(sview.hasEntry(coin1));
+      assert.strictEqual(mempool.hasCoin(coin1.hash, coin1.index), true);
+      assert.strictEqual(mempool.isSpent(coin1.hash, coin1.index), false);
+    }
+
+    await mempool.addTX(tx2);
+
+    {
+      const view = await mempool.getCoinView(tx1);
+      const sview = await mempool.getSpentView(tx1);
+
```

Comment:
nit: other blocks dont have newline here, but i dont care

Reply:
why aren't you checking the tx1 view right after adding tx1 to mempool? You add tx2 first then check for `dummyCoin`?

Reply:
sview/view checks for dummyCoin can be done before tx2 is added, but that block mainly checks hasCoin and isSpend for the coin. The checks you are talking is the next block.